src/pages/ShoppingList.js

```diff
@@ -35,43 +64,85 @@ const ShoppingList = ({ token }) => {
     useEffect(() => {
         getShoppingList();
     }, []);
+
     const lessThan24Hours = date => {
         const formattedDate = parseInt(moment(date).format());
         const newDate = moment(Date.now());
         const lastPurchase = moment(formattedDate);
         return lastPurchase.diff(newDate, 'hours') < 24;
     };
 
-    const handleCheck = e => {
-        let db = fb.firestore();
-        let tokenRef = db.collection(userToken).doc(e.target.name);
-        let dataCheck = {
-            isChecked: e.target.checked,
-            lastPurchaseDate: moment(Date.now()).format(),
-        };
-        tokenRef.update(dataCheck).then(function() {
-            getShoppingList();
-        });
+    const HOURS24 = 86400; //24 hours in seconds
+
+    const handleCheck = (e,item) => {
+        if (!(item.lastPurchaseDate == null)) {
+            let lastEstimate;
+            item.nextPurchaseDate
+              ? (lastEstimate = item.nextPurchaseDate)
+              : (lastEstimate = item.timeFrame);
+            let lastPurchaseDate = item.lastPurchaseDate;
+
+            let today = moment(Date.now())
+            let lastPurchase = moment(lastPurchaseDate);
+            let latestInterval = today.diff(lastPurchase, 'days');
+
+            let db = fb.firestore();
+            let nextPurchaseDate = calculateEstimate(
+              item.lastEstimate,
+              latestInterval,
+              item.numOfPurchases
+            );
+            db.collection(userToken)
+              .doc(e.target.name)
+              .update({
+                lastPurchaseDate,
+                numOfPurchases: item.isChecked == false ? (item.numOfPurchases || 0) + 1 : item.numOfPurchases,
+                latestInterval,
```

Comment:
Could you pull out `item.isChecked == false ? (item.numOfPurchases || 0) + 1 : item.numOfPurchases` into a variable so you could just use it in both places?  It'll make it easier reading it going forward, I think.